modules/RTC/TraceablePeerConnection.js

```diff
@@ -1466,30 +1333,54 @@ TraceablePeerConnection.prototype._updateAv1DdHeaders = function(description) {
  */
 TraceablePeerConnection.prototype.addTrack = function(track, isInitiator = false) {
     const rtcId = track.rtcId;
-    let transceiver;
 
-    logger.info(`${this} adding ${track}`);
     if (this.localTracks.has(rtcId)) {
-
         return Promise.reject(new Error(`${track} is already in ${this}`));
     }
 
-    this.localTracks.set(rtcId, track);
+    logger.info(`${this} adding ${track}`);
     const webrtcStream = track.getOriginalStream();
+    const mediaStreamTrack = track.getTrack();
+    let transceiver;
+
+    if (isInitiator) {
+        const streams = [];
 
-    try {
-        transceiver = this.tpcUtils.addTrack(track, isInitiator);
-    } catch (error) {
-        logger.error(`${this} Adding track=${track} failed: ${error?.message}`);
+        webrtcStream && streams.push(webrtcStream);
 
-        return Promise.reject(error);
+        // Use pc.addTransceiver() for the initiator case when local tracks are getting added
+        // to the peerconnection before a session-initiate is sent over to the peer.
+        const transceiverInit = {
+            direction: MediaDirection.SENDRECV,
+            streams,
+            sendEncodings: []
+        };
+
+        if (!browser.isFirefox()) {
+            transceiverInit.sendEncodings = this.tpcUtils.getStreamEncodings(track);
+        }
+
+        try {
+            transceiver = this.peerconnection.addTransceiver(mediaStreamTrack, transceiverInit);
+        } catch (error) {
```

Comment:
Suggestion: make this function async so you don't need to do this, execptions will reject the promsie already.